src/appleseed/renderer/kernel/rendering/final/uniformpixelrenderer.cpp

```diff
@@ -151,116 +133,55 @@ namespace
-                            instance,                   // number of samples
-                            instance);                  // initial instance number -- end of sequence
-
-                        // Render the sample.
-                        ShadingResult shading_result(aov_count);
-                        m_sample_renderer->render_sample(
-                            sampling_context,
-                            pixel_context,
-                            sample_position,
-                            aov_accumulators,
-                            shading_result);
-
-                        // Update sampling statistics.
-                        m_total_sampling_dim.insert(sampling_context.get_total_dimension());
-
-                        // Merge the sample into the framebuffer.
-                        if (shading_result.is_valid())
-                        {
-                            framebuffer.add(
-                                static_cast<float>(s.x - pi.x + pt.x),
-                                static_cast<float>(s.y - pi.y + pt.y),
-                                shading_result);
-                        }
-                        else signal_invalid_sample();
-                    }
-                }
+                // Generate a uniform sample in [0,1)^2.
+                const Vector2f s =
+                    m_sample_count > 1 || m_params.m_force_aa
+                        ? sampling_context.next2<Vector2f>()
+                        : Vector2f(0.5f);
+
+                // Sample the pixel filter.
+                const auto& filter_table = frame.get_filter_sampling_table();
+                const Vector2d pf(
+                    static_cast<double>(filter_table.sample(s[0]) + 0.5f),
+                    static_cast<double>(filter_table.sample(s[1]) + 0.5f));
+
+                // Compute the sample position in NDC.
+                const Vector2d sample_position = frame.get_sample_position(pi.x + pf.x, pi.y + pf.y);
+
+                // Create a pixel context that identifies the pixel and sample currently being rendered.
+                const PixelContext pixel_context(pi, sample_position);
+
+                // Render the sample.
+                ShadingResult shading_result(aov_count);
+                SamplingContext child_sampling_context(sampling_context);
+                m_sample_renderer->render_sample(
+                    child_sampling_context,
+                    pixel_context,
+                    sample_position,
+                    aov_accumulators,
+                    shading_result);
+
+                // Update sampling statistics.
+                m_total_sampling_dim.insert(child_sampling_context.get_total_dimension());
+
+                // Merge the sample into the framebuffer.
+                if (shading_result.is_valid())
+                    framebuffer.add(Vector2u(pt), shading_result);
+                else signal_invalid_sample();
```

Comment:
```
else
    signal_invalid_sample();
```
isn't better?

Reply:
The way we currently format things is as @est77 did it. This will change when we'll switch to clang-format.